ompi/mpi/fortran/use-mpi-f08/cdesc/bsend_init_cdesc.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2005 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2011-2012 Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2015-2019 Research Organization for Information Science
+ *                         and Technology (RIST).  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "ompi/mpi/fortran/mpif-h/bindings.h"
+#include "ompi/mpi/fortran/use-mpi-f08/cdesc/bindings.h"
+#include "ompi/mpi/fortran/base/constants.h"
+
+
+void ompi_bsend_init_cdesc(CFI_cdesc_t* x, MPI_Fint *count, MPI_Fint *datatype, MPI_Fint *dest, MPI_Fint *tag, MPI_Fint *comm, MPI_Fint *request, MPI_Fint *ierr)
+{
+    int c_ierr;
+    MPI_Datatype c_datatype, c_type = PMPI_Type_f2c(*datatype);
+    MPI_Request c_req;
+    MPI_Comm c_comm;
+    void *buf = x->base_addr;
+    int c_count =  OMPI_FINT_2_INT(*count);
+
+    c_comm = PMPI_Comm_f2c (*comm);
+
+    c_datatype = c_type;
+    if (x->rank != 0 && !CFI_is_contiguous(x)) {
+        c_ierr = ompi_cdesc_create_datatype(x, c_count, c_type, &c_datatype);
+        if (MPI_SUCCESS != c_ierr) {
+            if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
```

Comment:
about that ...

there is a lot of copy/paste from `mpifh` (and there are no `{}` here)

but the real question is do we really need to test `if (NULL != ierr)` ?
`ierr` cannot be `NULL` if invoked from `mpif-h` or `usempi`.
If the Fortran parameter is omitted in `usempif08`, then we still invoke this subroutine with an address on the stack.

Reply:
Good point.  I think you're right -- we might as well remove all the `if (NULL != ierr)` from this layer because the upper layer guarantees to call with a valid (non-NULL) `ierr`.